sycl/test/check_device_code/vector/vector_math_ops.cpp

```diff
@@ -155,6 +226,49 @@ SYCL_EXTERNAL auto TestAdd(vec<half, 3> a, vec<half, 3> b) { return a + b; }
+// CHECK-ARR-STORAGE-NEXT:  entry:
+// CHECK-ARR-STORAGE-NEXT:    [[REF_TMP_I_I:%.*]] = alloca float, align 4
+// CHECK-ARR-STORAGE-NEXT:    [[REF_TMP1_I:%.*]] = alloca %"class.sycl::_V1::ext::oneapi::bfloat16", align 2
+// CHECK-ARR-STORAGE-NEXT:    [[REF_TMP3_I:%.*]] = alloca %"class.sycl::_V1::ext::oneapi::bfloat16", align 2
+// CHECK-ARR-STORAGE-NEXT:    tail call void @llvm.experimental.noalias.{{.*}}
+// CHECK-ARR-STORAGE-NEXT:    call void @llvm.lifetime.start.p0(i64 2, ptr nonnull [[REF_TMP1_I]])
+// CHECK-ARR-STORAGE-NEXT:    call void @llvm.lifetime.start.p0(i64 2, ptr nonnull [[REF_TMP3_I]])
+// CHECK-ARR-STORAGE-NEXT:    [[REF_TMP1_ASCAST_I:%.*]] = addrspacecast ptr [[REF_TMP1_I]] to ptr addrspace(4)
+// CHECK-ARR-STORAGE-NEXT:    [[REF_TMP3_ASCAST_I:%.*]] = addrspacecast ptr [[REF_TMP3_I]] to ptr addrspace(4)
+// CHECK-ARR-STORAGE-NEXT:    [[REF_TMP_ASCAST_I_I:%.*]] = addrspacecast ptr [[REF_TMP_I_I]] to ptr addrspace(4)
+// CHECK-ARR-STORAGE-NEXT:    br label [[FOR_COND_I:%.*]]
+// CHECK-ARR-STORAGE:       for.cond.i:
+// CHECK-ARR-STORAGE-NEXT:    [[I_0_I:%.*]] = phi i64 [ 0, [[ENTRY:%.*]] ], [ [[INC_I:%.*]], [[FOR_BODY_I:%.*]] ]
+// CHECK-ARR-STORAGE-NEXT:    [[CMP_I:%.*]] = icmp ult i64 [[I_0_I]], 3
+// CHECK-ARR-STORAGE-NEXT:    br i1 [[CMP_I]], label [[FOR_BODY_I]], label [[_ZN4SYCL3_V1PLERKNS0_3VECINS0_3EXT6ONEAPI8BFLOAT16ELI3EEES7__EXIT:%.*]]
+// CHECK-ARR-STORAGE:       for.body.i:
+// CHECK-ARR-STORAGE-NEXT:    call void @llvm.experimental.noalias.{{.*}}
+// CHECK-ARR-STORAGE-NEXT:    call void @llvm.experimental.noalias.{{.*}}
+// CHECK-ARR-STORAGE-NEXT:    [[ARRAYIDX_I_I_I_I_I:%.*]] = getelementptr inbounds [4 x i16], ptr [[A]], i64 0, i64 [[I_0_I]]
+// CHECK-ARR-STORAGE-NEXT:    [[TMP0:%.*]] = load i16, ptr [[ARRAYIDX_I_I_I_I_I]], align 2, {{.*}}
+// CHECK-ARR-STORAGE-NEXT:    store i16 [[TMP0]], ptr [[REF_TMP1_I]], align 2, {{.*}}
+// CHECK-ARR-STORAGE-NEXT:    call void @llvm.experimental.noalias.{{.*}}
+// CHECK-ARR-STORAGE-NEXT:    call void @llvm.experimental.noalias.{{.*}}
+// CHECK-ARR-STORAGE-NEXT:    [[ARRAYIDX_I_I_I_I10_I:%.*]] = getelementptr inbounds [4 x i16], ptr [[B]], i64 0, i64 [[I_0_I]]
+// CHECK-ARR-STORAGE-NEXT:    [[TMP1:%.*]] = load i16, ptr [[ARRAYIDX_I_I_I_I10_I]], align 2, {{.*}}
+// CHECK-ARR-STORAGE-NEXT:    store i16 [[TMP1]], ptr [[REF_TMP3_I]], align 2, {{.*}}
+// CHECK-ARR-STORAGE-NEXT:    call void @llvm.lifetime.start.p0(i64 4, ptr nonnull [[REF_TMP_I_I]]), {{.*}}
+// CHECK-ARR-STORAGE-NEXT:    [[CALL_I_I_I_I:%.*]] = call spir_func noundef float @__devicelib_ConvertBF16ToFINTEL(ptr addrspace(4) noundef align 2 dereferenceable(2) [[REF_TMP1_ASCAST_I]]) #[[ATTR8:[0-9]+]], {{.*}}
```

Comment:
Why don't we store BFloat16 as _Float16, just like half, instead of uint16? That would reduce calls to _ConvertBF16ToFINTEL and _ConvertFToBF16INTEL?